docs/DesignDocs/NotesOnAttestationAPI.md

```diff
@@ -0,0 +1,366 @@
+function `oe_get_evidence()`.
+
+Among all the parameters for the OE SDK attestation API
+function `oe_get_evidence()`,
+two of them are a list of custom claims (`custom_claims`) and an optional
+parameters blob(`opt_params`). While in general the fields in these two
+parameters fit the definition of the term "Claims" defined in the
+[RATS document](https://tools.ietf.org/html/draft-ietf-rats-architecture),
+in the context of the function `oe_get_evidence()`, these two parameters
+have distinct semantics that might not be straightforward to application
+and plugin software developers.
+
+To help clarify what goes into the custom claims (`custom_claims`) and
+what goes into the optional parameters (`opt_params`) in the function
+`oe_get_evidence()`, it would be good to articulate the semantics of these
+two entities.
+
+- Custom claims have well-defined semantics:
+  - Custom claims can be any flat array of bytes supplied by attester
+  applications.
+  - Custom claims are only meaningful to attester / verifier applications.
+  They are not inspected or interpreted by the OE SDK or its plugins.
+  - They are always TEE agnostic
+  - Custom claims are included and signed in the evidence as clear text
+  by an attester for integrity protection.
+  On the verifier side, they are verified by a verifier plugin
+  as part of the evidence verification, and presented to the verifier
+  application for processing.
+    - Secrets shall never be included as a custom claim. Instead a hash of
+    a secret can be included by an attester as a custom claim in its evidence,
+    and passed to a verifier with integrity protection. The secret itself
+    has to be passed from the attester to the verifier with confidentiality
+    protection. Integrity of the secret can be verified by the verifier
+    using the hash held as a custom claim in the evidence.
+  - Custom claims can cover attester instance identity or
+  properties.
+    - This type of custom claims are created and maintained by the attester
+    and not received from a verifier.
+    - This type of claims are passed to verifiers with integrity protection
+    as part of the evidence, so verifiers can use the claims to learn the
+    identity and properties for the attester and for secure communication
+    with the latter.
+    - The public key of an asymmetric key-pair generated by an attester,
+    serving as an unique public identity of the attester,
+    is a good example of a custom claim.
+  - Custom claims can also be created by a verifier application and passed to
+  an attester for inclusion in the evidence.
+    - This type of claims allows the attester to make assertions about
+    its evidence to the targeted verifier.
+    - A good example of this type of claims is a nonce generated by a
```

Comment:
FYI Text is ok, but I don't think it's actually true in general.   A nonce (e.g., the size in bytes of the nonce can be fixed and vary by format or by TEE, or one format or TEE might require a timestamp whereas another requires a random number) can be format-specific and TEE-specific.   So this text is just one way of doing a nonce, and in that sense is ok, but nonces can also be done as optional parameters.   Again it's ok to not change the text if this is understood.

Reply:
The nonce is used as example. I think the sub-bullets could be confusing, and I'll delete them.